ratpack-kafka-producer/src/test/groovy/smartthings/ratpack/kafka/KafkaProducerServiceSpec.groovy

```diff
@@ -156,4 +156,154 @@ class KafkaProducerServiceSpec extends Specification {
 		assert result.throwable.getClass() == IllegalStateException
 		assert result.throwable.getMessage() == 'KafkaProducer is currently not available.'
 	}
+
+	def "Can send a message with optional properties"() {
```

Comment:
You should add test cases to check those properties are set properly in producer service by setting it first then retrieve and compare.
To test a list of properties with different values and combinations, you can utilise Spock data table and unroll.